src/server/game/Entities/Player/Player.cpp

```diff
@@ -24669,6 +24669,8 @@ bool Player::HasItemFitToSpellRequirements(SpellInfo const* spellInfo, Item cons
                     if (!item || !item->IsFitToSpellRequirements(spellInfo))
                         return false;
                 }
+
+                return true;
```

Comment:
according to the logic of the other cases in that function and the functionname, it should be:
```cpp
for (uint8 i : {EQUIPMENT_SLOT_HEAD, EQUIPMENT_SLOT_SHOULDERS, EQUIPMENT_SLOT_CHEST, EQUIPMENT_SLOT_WAIST, EQUIPMENT_SLOT_LEGS, EQUIPMENT_SLOT_FEET, EQUIPMENT_SLOT_WRISTS, EQUIPMENT_SLOT_HANDS})
{
    if (Item* item = GetUseableItemByPos(INVENTORY_SLOT_BAG_0, i))
        if (item != ignoreItem && item->IsFitToSpellRequirements(spellInfo))
            return true;
}
```

Reply:
This is a special case where all slots must have a specific item type, so the logic here is good, it was only missing a specific return true; at the end

Another spell example : http://www.wowhead.com/spell=86538

Reply:
ok, got it
about your known issues, try this:
```diff
diff --git a/src/server/game/Entities/Player/Player.cpp b/src/server/game/Entities/Player/Player.cpp
index 5c28e7b72b..240d0668b8 100644
--- a/src/server/game/Entities/Player/Player.cpp
+++ b/src/server/game/Entities/Player/Player.cpp
@@ -24666,9 +24666,11 @@ bool Player::HasItemFitToSpellRequirements(SpellInfo const* spellInfo, Item cons
                 for (uint8 i : {EQUIPMENT_SLOT_HEAD, EQUIPMENT_SLOT_SHOULDERS, EQUIPMENT_SLOT_CHEST, EQUIPMENT_SLOT_WAIST, EQUIPMENT_SLOT_LEGS, EQUIPMENT_SLOT_FEET, EQUIPMENT_SLOT_WRISTS, EQUIPMENT_SLOT_HANDS})
                 {
                     Item* item = GetUseableItemByPos(INVENTORY_SLOT_BAG_0, i);
-                    if (!item || !item->IsFitToSpellRequirements(spellInfo))
+                    if (!item || item == ignoreItem || !item->IsFitToSpellRequirements(spellInfo))
                         return false;
                 }
+
+                return true;
             }
 
             break;
```

Reply:
Worked indeed, thanks @joschiwald